R/pkg/R/DataFrame.R

```diff
@@ -303,8 +303,28 @@ setMethod("colnames",
 #' @rdname columns
 #' @name colnames<-
 setMethod("colnames<-",
-          signature(x = "DataFrame", value = "character"),
+          signature(x = "DataFrame"),
           function(x, value) {
+
+            # Check parameter integrity
+            if (class(value) != "character") {
+              stop("Invalid column names.")
+            }
+
+            if (length(value) != ncol(x)) {
+              stop(
+                "Column names must have the same length as the number of columns in the dataset.")
+            }
+
+            if (any(is.na(value))) {
+              stop("Column names cannot be NA.")
+            }
+
+            # Check if the column names have . in it
+            if (any(regexec(".", value, fixed=TRUE)[[1]][1] != -1)) {
```

Comment:
To clarify, I think having a specific test of colnames with iris will help to realize that this check [code](https://github.com/apache/spark/pull/11220/files#diff-508641a8bd6c6b59f3e77c80cdcfa6a9R324) is in place and should be removed, if and when SPARK-11976 is fixed. Otherwise only tests with str and with will be fixed. Does that make sense?